### PR TITLE
feat: fix carSalesRejection

### PR DESCRIPTION
**Pull request by Marine-Berthier**
Related to [this PR](https://github.com/carforyou/carforyou-dealerhub-web/pull/1357)

### CODE REVIEW

File: src/services/reporting/carSales.ts [hunk 1]
```diff
@@ -34,15 +34,12 @@ export const fetchCarSales = async ({
...
   } catch (error) {
-    return handleValidationError({
-      error,
-      options: { swallowErrors: true },
-    })
+    return handleValidationError(error)
+  }
+  return {
+    tag: "success",
+    result: null,
```
Comment (ufkaya) [inline src/services/reporting/carSales.ts:42]: Why did you move it outside the `try()`?

Comment (Marine-Berthier) [reply src/services/reporting/carSales.ts:42]: Because I tried to stick to the way we usually try / catch, but it can remain inside actually 👍

File: src/services/reporting/carSales.ts [hunk 2]
```diff
@@ -39,10 +39,7 @@ const envelopeRequest = async (handler): Promise<WithValidationError<null>> => {
       result: null,
     }
   } catch (error) {
-    return handleValidationError({
-      error,
-      options: { swallowErrors: true },
-    })
+    return handleValidationError(error)
```
Comment (ufkaya) [inline src/services/reporting/carSales.ts:42]: Are you sure that it is related to the `swallowErrors` option? How it is related?

Comment (Marine-Berthier) [reply src/services/reporting/carSales.ts:42]: When `swallowErrors` is true then the response is undefined. 
But to be honest I don't understand why. Here : https://github.com/carforyou/carforyou-api-client-pkg/blob/2de06f89567613bb5036ccf8090b0d01c9154908/src/base.ts#L207, our `error.name` is "ResponseError" so it should not matter

File: src/services/reporting/carSales.ts [hunk 3]
```diff
@@ -39,10 +39,7 @@ const envelopeRequest = async (handler): Promise<WithValidationError<null>> => {
       result: null,
     }
   } catch (error) {
-    return handleValidationError({
-      error,
-      options: { swallowErrors: true },
-    })
+    return handleValidationError(error, { swallowErrors: true })
```
Comment (Marine-Berthier) [inline src/services/reporting/carSales.ts:42]: @ufkaya I think I found the "real mistake". Before it was `handleValidationError({error, options :{ swallowErrors: true }})`
This would explain why e`rror.statusCode` was not enough and that I had to `error.error.statusCode`

Comment (ufkaya) [reply src/services/reporting/carSales.ts:42]: Nice! Good catch!

**Review by ufkaya — APPROVED** [review]
👏
